### ISSUE TITLE
Invalid regex warning from HA

### ISSUE THREAD

**Issue by piotrtobolski** [barleybobs/homeassistant-ecowater-softener#59]
I'm getting this warning in logs:

```
2024-06-07 23:41:49.917 WARNING (ImportExecutor_0) [py.warnings] /config/custom_components/ecowater_softener/coordinator.py:52: SyntaxWarning: invalid escape sequence '\)'
nextRecharge_re = "device-info-nextRecharge'\)\.html\('(?P<nextRecharge>.*)'"
```

**Comment by figorr** [comment]
I have just seen this warning at log when I restarted HA. It looks like there is a \ that could cause the conflict.
Maybe using this expression ...
`nextRecharge_re = r"device-info-nextRecharge'\)\.html\('(?P<nextRecharge>.*)'"`

**Comment by barleybobs** [comment]
Thanks @figorr that regex does seem to fix the issue. I've included it in [v3.4.1](https://github.com/barleybobs/homeassistant-ecowater-softener/releases/tag/v3.4.1).

Thanks 👍
